packages/runtime/src/next/client.ts

```diff
@@ -201,23 +230,51 @@ export class Makeswift {
     return response
   }
 
-  async getPages({
-    siteVersion = MakeswiftSiteVersion.Live,
-  }: {
-    siteVersion?: MakeswiftSiteVersion
-  } = {}): Promise<MakeswiftPage[]> {
-    const response = await this.fetch(`v3/pages`, siteVersion)
+  private getPagesInternal = async (
+    args: {
+      siteVersion?: MakeswiftSiteVersion
+      limit?: number
+      after?: string
+      sortBy?: 'title' | 'path' | 'createdAt' | 'updatedAt' | 'publishedAt'
+      sortDirection?: 'asc' | 'desc'
+      includeOffline?: boolean
+      pathPrefix?: string
+      locale?: string
+    } = {},
+  ) => {
```

Comment:
We should define function return types in addition to argument types to avoid TypeScript implicitly setting a return type